### DISCUSSION TITLE
请教 如何在自动生成的API基础上修改返回自定义的数据

### DISCUSSION THREAD

**Discussion by 18285160741** [zuihou/zuihou-admin-cloud#80] [02.提问]
自动生成的API都是基于单表的，但是很多时候一个列表需要同时查出多表的内容。
是否可以保留原有的增删改接口，只改变列表返回的数据实体为自定义实体呢（不想直接修改表对应的实体）。
我现在是直接新增了一个API，就只增加了个别字段，比如orgId对应的orgName之类的，感觉很麻烦且冗余。

实际举例：
设备Device（表里存的所属组织org_id）、组织org、自定义DeviceDTO（含Device所有字段、增加了组织名称orgName）
1、自动生成的设备controller
public class DeviceController extends SuperController<DeviceService, Long, Device, DevicePageQuery, DeviceSaveDTO, DeviceUpdateDTO> 

2、新增的列表API
public R<IPage<DeviceDTO>> list(@RequestBody @Validated PageParams<DevicePageQuery> params)

3、是否可以将原有的page接口返回的Device改为自定义DeviceDTO

请求指教，谢谢

**Comment by 18285160741** [comment]
粘贴的尖括号的内容怎么不见了，嘿嘿

2、新增的列表API
`public R<IPage<DeviceDTO>> list(@RequestBody @Validated PageParams<DevicePageQuery> params)`

**Reply by zuihou** [reply]
不可以， 其他解决方案：

1， 新增一个单独的查询接口

2，将org_id字段的类型改成 RemoteData ，并使用自动注入功能。 参考User类的 org 字段， 参考UserController的page 方法

**Reply by 18285160741** [reply]
好的，谢谢指点。
现在最多的是创建人、修改人要显示名称。有点头大。
